tutorials/native-debugging.md

```diff
@@ -71,4 +71,45 @@ settings append target.exec-search-paths <PATH>/objdir-android-opt/mozglue/build
 ```lldb
 b <file>.cpp:<line number>
 ```
-5. Once your breakpoints have been set, click the continue execution button to move beyond the `ElfLoader` breakpoint and your newly set native breakpoints should be hit. Debug as usual.
\ No newline at end of file
+5. Once your breakpoints have been set, click the continue execution button to move beyond the `ElfLoader` breakpoint and your newly set native breakpoints should be hit. Debug as usual.
+
+## Attaching debuggers to content and other child processes
+
+Internally, GeckoView has a multi-process architecture.  The main Gecko process lives in the main Android process, but content rendering and some other functions live in child processes.  This balances load, ensures certain critical security properties, and allows GeckoView to recover if content processes become unresponsive or crash.  However, it's generally delicate to debug child processes because they come and go.
+
+The general approach is to make the Java code in the child process that we want to debug wait for a Java debugger at startup, and then to connect such a Java debugger manually from the Android Studio UI.
```

Comment:
nit: We use the `you` phrasing everywhere so this use of `we` is a not in keeping with the rest of the document.